docs/source/preparing_datasets/basic_dataset_conversion.md

```diff
@@ -0,0 +1,330 @@
+| Encoding           | 'int'         | `Int`        | uses `numpy.int64`       |
+| Numpy Array        | 'ndarray'     | `NDArray`    | uses `numpy.ndarray`     |
+| Numpy Unsigned Int | 'uint8'       | `UInt8`      | uses `numpy.uint8`       |
+| Numpy Unsigned Int | 'uint16'      | `UInt16`     | uses `numpy.uint16`      |
+| Numpy Unsigned Int | 'uint32'      | `Uint32`     | uses `numpy.uint32`      |
+| Numpy Unsigned Int | 'uint64'      | `Uint64`     | uses  `numpy.uint64`     |
+| Numpy Signed Int   | 'int8'        | `Int8`       | uses  `numpy.int8`       |
+| Numpy Signed Int   | 'int16'       | `Int16`      | uses  `numpy.int16`      |
+| Numpy Signed Int   | 'int32'       | `Int32`      | uses  `numpy.int32`      |
+| Numpy Signed Int   | 'int64'       | `Int64`      | uses  `numpy.int64`      |
+| Numpy Float        | 'float16'     | `Float16`    | uses  `numpy.float16`    |
+| Numpy Float        | 'float32'     | `Float32`    | uses  `numpy.float32`    |
+| Numpy Float        | 'float64'     | `Float64`    | uses  `numpy.float64`    |
+| Numerical String   | 'str_int'     | `StrInt`     | stores in UTF-8          |
+| Numerical String   | 'str_float'   | `StrFloat`   | stores in UTF-8          |
+| Numerical String   | 'str_decimal' | `StrDecimal` | stores in UTF-8          |
+| Image              | 'pil'         | `PIL`        | raw PIL image            |
+| Image              | 'jpeg'        | `JPEG`       | PIL image as JPEG        |
+| Image              | 'png'         | `PNG`        | PIL image as PNG         |
+| Pickle             | 'pkl'         | `Pickle`     | arbitrary Python objects |
+| JSON               | 'json'        | `JSON`       | arbitrary data as JSON   |
+
+Here's an example where the field `x` is an image, and `y` is a class label, as an integer.
+<!--pytest.mark.skip-->
+```python
+column = {
+    'x': 'jpeg',
+    'y': 'int8',
+}
+```
+
+If the data type you need is not listed in the above table, then you can write your own data type class with `encode` and `decode` methods in it and patch it inside streaming. For example, let's say, you wanted to add a `complex128` data type (64 bits each for real and imaginary parts):
+
+<!--pytest.mark.skip-->
+```python
+import numpy as np
+from typing import Any
+
+from streaming.base.format.mds.encodings import Encoding, _encodings
+
+class Complex128(Encoding):
+
+    def encode(self, obj: Any) -> bytes:
+        return np.complex128(obj).tobytes()
+
+    def decode(self, data: bytes) -> Any:
+        return np.frombuffer(data, np.complex128)[0]
+
+_encodings['complex128'] = Complex128
+```
+
+4. An optional shard `size_limit`, in bytes, for each *uncompressed* shard file. This defaults to 67 MB. Specify this as a number of bytes, either directly as an `int`, or a human-readable suffix:
+
+<!--pytest.mark.skip-->
+```python
+size_limit = 1024 # 1kB limit, as an int
+size_limit = "1kb" # 1kB limit, as a human-readable string
```

Comment:
we are like 95% standardized on single quotes in this PR i think

Reply:
will hunt down the others lol

Reply:
keeping the ones in JSON tho